### PR TITLE
OAK-9461: dynamic boost lite

### PR DESCRIPTION
**Pull request by fabriziofortino**
The initial idea was to change `oak-lucene` since the sparse index issue affects lucene only. The query part is shared in `oak-search` though. Because of that, I have introduced the new sys property `oak.search.dynamicBoostLite` that accepts a list of comma-separated types where the lite mode can be enabled. In this way, the dynamic boost lite can be enabled for some types and not for others. The sys property can be overridden by the `dynamicBoostLite` property at index definition level.

### CODE REVIEW

**Review by thomasmueller — APPROVED** [review]
Some unrelated changes, but I think that's fine.

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/dynamicBoost/DynamicBoostTest.java [hunk 1]
```diff
@@ -128,6 +128,20 @@ protected ContentRepository createRepository() {
                         "]", log);
     }
 
+    @Test public void withDynamicBoostLite() throws Exception {
+        NodeTypeRegistry.register(root, toInputStream(ASSET_NODE_TYPE), "test nodeType");
+        Tree props = createIndex("dam:Asset", true);
+        Tree pt = createNodeWithType(props, "predictedTags", UNSTRUCTURED);
+        pt.setProperty("name", "jcr:content/metadata/predictedTags/.*");
+        pt.setProperty("isRegexp", true);
+        pt.setProperty("dynamicBoost", true);
```
Comment (chibulcu) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/dynamicBoost/DynamicBoostTest.java:138]: `dynamicBoostLite ` ?

File: oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/IndexDefinition.java [hunk 2]
```diff
@@ -583,10 +609,6 @@ public int getMaxExtractLength() {
         return maxExtractLength;
     }
 
-    public String getScorerProviderName() {
-        return scorerProviderName;
-    }
-
```
Comment (chibulcu) [inline oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/IndexDefinition.java:589]: unused?

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/dynamicBoost/DynamicBoostTest.java (hunk 1 above)
Comment (fabriziofortino) [reply oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/dynamicBoost/DynamicBoostTest.java:138]: no, here `dynamicBoost` is correct. `dynamicBoostLite` is something at index definition level.

File: oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/IndexDefinition.java (hunk 2 above)
Comment (fabriziofortino) [reply oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/IndexDefinition.java:589]: yes, I have included a number of unrelated code improvements in this separate commit https://github.com/oak-indexing/jackrabbit-oak/pull/508/commits/3fc57cbf83b74987ede84d7e169e3ea1879b8f83

**Review by averma21 — APPROVED** [review]
LGTM. I think we should also mention somewhere that for dynamicBoostLite to work, similarity tags must be enabled on the field (which is generally the case but we could explicitly mention it just to be more clear).

**Comment by fabriziofortino** [comment]
@averma21 I have added more documentation on the point you have raised. Please review.

**Pull request closed without merging**